File: backend/src/Asnan.Domain/Enums/NotificationCategory.cs
namespace Asnan.Domain.Enums;

public enum NotificationCategory
{
    /// <summary>Scheduled/cancelled/starting-soon. Transactional — non-disable-able (ARCHITECTURE.md §10).</summary>
    AppointmentUpdates = 1,

    /// <summary>Payment succeeded/failed, refund completed. Transactional/security-adjacent — non-disable-able.</summary>
    PaymentUpdates = 2,

    /// <summary>Appointment-reminder pushes (Milestone 6's ReminderSchedulingService). Disableable.</summary>
    Reminders = 3,

    /// <summary>New chat message while the recipient is offline. Disableable.</summary>
    ChatMessages = 4,

    /// <summary>A doctor's availability changed. Disableable.</summary>
    DoctorAvailability = 5,
}

File: mobile/lib/features/notifications/domain/notification_category.dart
/// Mirrors the backend's `NotificationCategory` enum (int-valued in JSON).
enum NotificationCategory {
  appointmentUpdates(1, 'Appointment updates'),
  paymentUpdates(2, 'Payment updates'),
  reminders(3, 'Appointment reminders'),
  chatMessages(4, 'Chat messages'),
  doctorAvailability(5, "Doctor availability changes");

  const NotificationCategory(this.value, this.label);

  final int value;

  /// Display label for the preferences screen.
  final String label;

  static NotificationCategory fromValue(int value) => NotificationCategory.values.firstWhere(
        (category) => category.value == value,
        orElse: () => throw ArgumentError('Unknown NotificationCategory value: $value'),
      );
}
